### ISSUE TITLE
Compute methods aren't called until a reactive changes

### ISSUE THREAD

**Issue by darrenburns** [Textualize/textual#1227]
The following test fails because `app.double_number` is still `None`. It hasn't been computed yet because no reactive attributes have changed yet.

When initialising the `reactive` attrs, we could leave computed ones to the end.

*However*, this leaves another gotcha that means computed reactive attrs can't really depend on other computed reactive attrs, since if there's a dependency in the order they're computed, we'd be unaware of it.

```python
async def test_reactive_compute_first_time_set():
    class ReactiveComputeFirstTimeSet(App):
        number = reactive(1)
        double_number = reactive(None)

        def compute_double_number(self):
            return self.number * 2

    app = ReactiveComputeFirstTimeSet()
    async with app.run_test():
        await asyncio.sleep(.2)  # TODO: We sleep here while issue#1218 is open
        assert app.double_number == 2  # FAILS HERE!! The value is None!
```

**Comment by willmcgugan** [comment]
Computes are called when reactives change, which is by design. But this could be be considered astonishing behavior.

Perhaps on the very first time a compute is referenced, it should invoke the compute method if it hasn't yet been calculated.

The order of computes is a good point. How about we document that the computes are called in the order they are defined? Which I think may actually be what happens currently.

This might require a discussion. Suggest we go over it when @darrenburns is back in the office.

**Comment by rodrigogiraoserrao** [comment]
Closed by #1558
